### ISSUE TITLE
Special processing for Firestone firmware update from v810 to v820

### ISSUE THREAD

**Issue by gurevichmark** [xcat2/xcat-core#2908]
This is a request from BMC firmware team (Jayesh Patel).

Special steps need to be taken when firmware update is performed on Firestone machine going from version 810 to 820 or from 820 to 810. Instead of `hpm upgrade <file> force` that we do today, the following need to be done:

```
hpm upgrade <file> component 0 force
hpm upgrade <file> component 1 force
```
Wait for BMC to reboot (wait for `00` ready code for a period of time), then
```
hpm upgrade <file> component 2 force
``` 
Jayesh said the step 2 above will trigger BMC reboot. He also said we should issue a reboot again after step 3 above.

To check the current installed version `fru print 47` can be used, one of the returned lines is
`Product Version       : IBM-firestone-ibm-OP8_v1.12_2.73`
The format of that string can be used to figure out 810 or 820, `v1.xx_y.zz`, if y==1 it is 810, if y==2, it is 820

To check the version of the .htm file:
```
bash-4.1$ cat -v 8335_820.1642.20170405o_update.hpm | grep FW_DESC
FW_DESC=8335 SRC BUILD RR9 03242017
```
if it is 2016 then it is Op810 for firestone
if it is 2017 then it is Op820 for firestone

All Op810 drivers:  http://pfd.austin.ibm.com/releasenotes/indexop810.html 
All Op820 drivers: http://pfd.austin.ibm.com/releasenotes/indexop820.html

**Comment by daniceexi** [comment]
@gurevichmark do we need to reflect the change in our ``rflash`` command?

**Comment by gurevichmark** [comment]
Yes, this issue was opened for the purpose of modifying `rflash` command.

**Comment by gurevichmark** [comment]
Fixed by PR #2964

**Comment by whowutwut** [comment]
Modified the title to use external naming instead of internal code work for the hardware.
